src/Pact/Types/Type.hs

```diff
@@ -278,7 +310,9 @@ data Type v =
   , _tySchemaType :: Type v
   , _tySchemaPartial :: SchemaPartial } |
   TyFun { _tyFunType :: FunType v } |
-  TyUser { _tyUser :: v }
+  TyUser { _tyUser :: v } |
+  TyModSpec
```

Comment:
Note here on naming: while the KIP and #813 refer to "module references", in the type constructor it made more sense to think of them as "module specifications" since that's kind of the "what" and "why" -- "specify a module by enumerating the interfaces it must support". 

If `TyModRef` or something is preferred happy to change it.

Reply:
Especially since a type _is_ a specification, perhaps this name is too "literal". However it was helpful when working out unification below.

Reply:
I think we should adhere to the KIP. I don't see too much value in introducing another phrase to keep track of